File: 프로그래머스/lv2/17680. ［1차］ 캐시/［1차］ 캐시.py
def solution(cacheSize, cities):
    answer = 0
    stack = []
    if cacheSize == 0:
        answer += len(cities) * 5
        return answer
    for i in cities:
        i = i.upper()
        if i in stack:
            answer += 1
            stack.pop(stack.index(i))
        else:
            answer += 5
            if len(stack) >= cacheSize:
                stack.pop(0)
        stack.append(i)

    return answer

File: 프로그래머스/lv2/17680. ［1차］ 캐시/［1차］ 캐시.java
import java.util.*;

class Solution {
    public int solution(int cacheSize, String[] cities) {
        int answer = 0;
        if (cacheSize == 0) {
            return cities.length * 5;
        }
        LinkedList<String> q = new LinkedList<>();
        for (String city : cities) {
            city = city.toLowerCase();
            if (q.remove(city)) {
                answer += 1;
            }
            else {
                answer += 5;
                if (q.size() >= cacheSize) {
                    q.remove(0);
                }
            }
            q.add(city);
        }
        return answer;
    }
}